### PR TITLE
fix: make linkage truly infallible

### PR DESCRIPTION
**Pull request by Gankra**
also some driveby cleanups

### CODE REVIEW

File: cargo-dist/src/build/mod.rs [hunk 1]
```diff
@@ -215,7 +215,7 @@ impl BuildExpectations {
             });
             linkage
         } else {
-            determine_linkage(src_path, target)?
+            determine_linkage(src_path, target)
```
Comment (ashleygwilliams) [inline cargo-dist/src/build/mod.rs:218]: this is the diff i was looking for haha

File: cargo-dist/src/errors.rs [hunk 2]
```diff
@@ -303,11 +303,8 @@ pub enum DistError {
     },
 
     /// Linkage report can't be run for this target
-    #[error("unable to run linkage report for this type of binary; file was: {file_name:?}")]
-    LinkageCheckUnsupportedBinary {
-        /// The object's file name
-        file_name: Option<String>,
-    },
+    #[error("unable to run linkage report for this type of binary")]
+    LinkageCheckUnsupportedBinary,
```
Comment (ashleygwilliams) [inline cargo-dist/src/errors.rs:307]: i do think maintaining the name of the file would be useful

**Review by ashleygwilliams — COMMENTED** [review]
in general looks good, brought up in discord but does seem strange to group linkage and signing to me, but this might be because i am missing context?

File: cargo-dist/src/errors.rs (hunk 2 above)
Comment (Gankra) [reply cargo-dist/src/errors.rs:307]: The warning all of these flow into handles reporting the filename so it ends up being redundant noise

**Pull request merged**